### ISSUE TITLE
[1.10.3] Metals stopped working with the embedded launcher jar

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#7792]
## steps

Invoke BSP server as specified in bsp JSON that we generated.

## problem
             
```
2024.10.21 14:06:01 INFO  BSP server: [info] server was not detected. starting an instance
2024.10.21 14:06:01 INFO  BSP server: [warn] server is started using sbt-launch jar directly
2024.10.21 14:06:01 INFO  BSP server: [warn] this is not the recommended way: .sbtopts and .jvmopts files are not loaded and SBT_OPTS is ignored
2024.10.21 14:06:01 INFO  BSP server: [warn] either upgrade sbt to its latest version or make sure it is accessible from $PATH, and run 'sbt bspConfig'
2024.10.21 14:06:01 INFO  BSP server: Unrecognized option: --server
2024.10.21 14:06:01 INFO  BSP server: Error: Could not create the Java Virtual Machine.
2024.10.21 14:06:01 INFO  BSP server: Error: A fatal exception has occurred. Program will exit.
```

## notes

Likely this is caused by #7775

_Originally posted by @tgodzik in https://github.com/sbt/sbt/pull/7775#discussion_r1808995003_

**Comment by Friendseeker** [comment]
c.c. https://github.com/jline/jline3

> The [Jansi](https://github.com/fusesource/jansi) library is a library specialized in supporting ANSI sequences in terminals. Historically, the JNI methods used by JLine were provided by Jansi. In order to minimize the maintenance cost, Jansi has been merged into JLine 3.25.
> 
> This provider has been deprecated in 3.26 in favor of the JNI provider.

Guess we just remove it

**Comment by tgodzik** [comment]
Looks like this might have been caused by using `--sbt-launch-jar` option we had in tests (which I don't think is neccessay)

Checking it here https://github.com/scalameta/metals/pull/6876
